### PR TITLE
Add some missing LSP protocol definitions

### PR DESCRIPTION
**Pull request by dibarbet**
Used by xaml

### CODE REVIEW

File: src/LanguageServer/Protocol/Protocol/Internal/VSInternalDocumentOnAutoInsertRegistrationOptions.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+// See the LICENSE file in the project root for more information.
+
+using System.Text.Json.Serialization;
+
+namespace Roslyn.LanguageServer.Protocol;
+
+/// <summary>
+/// Class representing the registration options for on auto insert support.
+/// </summary>
+internal class VSInternalDocumentOnAutoInsertRegistrationOptions : VSInternalDocumentOnAutoInsertOptions, ITextDocumentRegistrationOptions, IStaticRegistrationOptions
```
Comment (mgoertz-msft) [inline src/LanguageServer/Protocol/Protocol/Internal/VSInternalDocumentOnAutoInsertRegistrationOptions.cs:12]: I noticed that `VSInternalDocumentOnAutoInsertOptions` doesn't use `JsonPropertyName` but still `DataMember`. We need to update all of those as well so they serialize correctly.

Comment (dibarbet) [reply src/LanguageServer/Protocol/Protocol/Internal/VSInternalDocumentOnAutoInsertRegistrationOptions.cs:12]: @mgoertz-msft hmm, it looks correct to me?
https://github.com/dotnet/roslyn/blob/main/src/LanguageServer/Protocol/Protocol/Internal/VSInternalDocumentOnAutoInsertOptions.cs

Comment (mgoertz-msft) [reply src/LanguageServer/Protocol/Protocol/Internal/VSInternalDocumentOnAutoInsertRegistrationOptions.cs:12]: Hah, I see what happened. I made sure I was looking at main but I ended up in your fork :)

https://github.com/dibarbet/roslyn/blob/main/src/Features/LanguageServer/Protocol/Protocol/Internal/VSInternalDocumentOnAutoInsertOptions.cs

Comment (mgoertz-msft) [reply src/LanguageServer/Protocol/Protocol/Internal/VSInternalDocumentOnAutoInsertRegistrationOptions.cs:12]: That's what I get for using the go to definition feature in a PR :)

**Review by mgoertz-msft — APPROVED** [review]

**Review by JoeRobich — APPROVED** [review]

**Pull request merged**